src/cpu.h

```diff
@@ -1,55 +1,64 @@
-#ifndef CPU_H
-#define CPU_H
-#include <stdint.h>
+#ifndef YOBEMAG_CPU_H
+#define YOBEMAG_CPU_H
 
-typedef struct cpu {
-    uint8_t H;
-    uint8_t L;
+#include <stdint.h>
 
-    uint8_t D;
-    uint8_t E;
+typedef enum flag {
+	/*
+	 * Carry flag
+	 */
+	C_FLAG = 4,
+	/*
+	 * Half carry flag
+	 */
+	H_FLAG = 5,
+	/*
+	 * Subtract flag
+	 */
+	N_FLAG = 6,
+	/*
+	 * Zero flag
+	 */
+	Z_FLAG = 7,
+} Flag;
 
-    uint8_t B;
-    uint8_t C;
+typedef union Register {
+	struct {
+		uint8_t high; // A, D, D, H
+		uint8_t low; // F, C, E, L
+	} bytes;
+	uint16_t word;
+} Register;
 
-    uint8_t A;
-    uint8_t F;
+typedef struct cpu {
+	Register HL;
+	Register DE;
+	Register BC;
+	Register AF;
 
     uint16_t SP;
     uint16_t PC;
+
     uint16_t cycle_count;
+
     uint8_t opcode;
-} cpu;
+} Cpu;
 
-extern cpu c;
+extern Cpu cpu;
```

Comment:
Same as above. This is much more an aesthetic thing then anything else, but `Cpu` just looks wrong to me.

Reply:
I think it's fine, CPU could be confused with UPPER_CASE such as macros.

Reply:
It really is a shame that *_t is namespaced by POSIX.

Reply:
I would also argue for something else, it simply doesn't fit somehow :D
Any other suggestions?

Reply:
CentralProcessingUnit 😎

Reply:
I mean actually it won't break PascalCase if we use CPU 🤔 
I suppose I was a bit too quick before, it should be fine 🤔

Reply:
PascalCase is fine for me if acronyms are respected and written in uppercase as they suppose to be